### PR TITLE
Updates TRISA documentation with state workflows

### PR DESCRIPTION
**Pull request by bbengfort**
Updates the TRISA documentation with transfer workflows to better help VASPs implement the TRISA travel rule information exchange mechanism and to describe what Envoy is doing with Transaction states.

### CODE REVIEW

File: docs/content/api/api.en.md
Comment (bbengfort) [inline docs/content/api/api.en.md:1]: The ` in the headers was rendering strangely, so I removed them.

File: docs/content/api/api.en.md [hunk 1]
```diff
@@ -57,15 +57,35 @@ message SecureEnvelope {
...
+    UNSPECIFIED = 0;  // the transfer state is unknown or not specified
+    STARTED = 1;      // this is the first message in the TRISA workflow
+    PENDING = 2;      // action is required by the sending party
+    REVIEW = 3;       // action is required by the recieving party (rarely used)
+    REPAIR = 4;       // some state of the travel rule exchange requires repair
+    ACCEPTED = 5;     // the travel rule exchange is accepted and waiting the transaction
+    COMPLETED = 6;    // the travel rule and on-chain transaction have been completed
+    REJECTED = 7;     // the travel rule exchange is rejected and should not proceed
+}
+```
```
Comment (bbengfort) [inline docs/content/api/api.en.md:76]: Added to docs

File: docs/content/api/api.en.md [hunk 2]
```diff
@@ -57,15 +57,35 @@ message SecureEnvelope {
...
+Address confirmation was initially described in the TRISA whitepaper as a mechanism to allow an originator VASP to establish that a beneficiary VASP has control of a crypto wallet address before sending transaction information with sensitive PII data.
+
+Currently, there are three proposed address confirmation methods being tested: simple, key/token, and on-chain. Simple involves a simple yes/no check with the counterparty. Key/Token requires the counterparty to decrypt a token encrypted with the crypto address public keys, and on-chain implements a simple Satoshi test.
+
+All three methods are available in the protocol buffers, but note that these are still being tested and developed, and may change at any time. No backward compatibility is guaranteed for these message types, nor is there any guarantee that any TRISA VASP will implement any of these confirmation methods.
```
Comment (bbengfort) [inline docs/content/api/api.en.md:85]: Updated the note about address confirmation.

File: docs/content/api/errors.en.md [hunk 3]
```diff
@@ -12,49 +12,52 @@ See the [error protocol buffers](https://github.com/trisacrypto/trisa/blob/main/
 
 ## TRISA Errors Codes, Messages, and Descriptions
```
Comment (bbengfort) [inline docs/content/api/errors.en.md:13]: Updated the error codes chart to ensure it was up to date with the latest protocol buffers.

File: docs/content/api/workflows.md
Comment (bbengfort) [inline docs/content/api/workflows.md:1]: Main docs for review including new Mermaid diagrams.

File: pkg/trisa/api/v1beta1/states.go [hunk 4]
```diff
@@ -6,6 +6,7 @@ const (
 	TransferStarted          = TransferState_STARTED
 	TransferPending          = TransferState_PENDING
 	TransferRepair           = TransferState_REPAIR
+    TransferReview           = TransferState_REVIEW
```
Comment (bbengfort) [inline pkg/trisa/api/v1beta1/states.go:9]: This is a state in Envoy, but wouldn't be used in an exchange; still I included it here just for completeness.

Comment (bbengfort) [reply pkg/trisa/api/v1beta1/states.go:9]: This is the "action required" state … but I do plan on changing it's name to review required.

**Review by daniellemaxwell — APPROVED** [review]
Thanks for keeping the docs updated! The workflows was especially helpful.

I made a minor spelling correction in the `api.proto` file. I had trouble attempting to generate the `api.pb.go` file, so it will need to be regenerated in order to apply the fix there as well.

File: docs/content/api/workflows.md [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    STARTED
+    autoAccept{{Review Required?}}
+    STARTED-->autoAccept
+    autoAccept -- Yes --> PENDING
+    autoAccept -- No --> ACCEPTED
+    accept{{Accept Transfer?}}
+    reject{{Retry and Fix?}}
+    PENDING-->accept
+    accept -- Yes --> ACCEPTED
+    accept -- No --> reject
```
Comment (daniellemaxwell) [inline docs/content/api/workflows.md:35]: Should `reject` be lowercase here? Just checking for consistency.

Comment (bbengfort) [reply docs/content/api/workflows.md:35]: Yes, sorry for the confusion - it's referencing the `reject` decision defined on Line 32 -- in the diagram it displays as "Retry and Fix?" -- it's a variable definition that is different than the "REJECTED" node.

**Pull request merged**